lib/Conversion/Utils/Utils.cpp

```diff
@@ -347,6 +347,100 @@ Value toPositiveValidDim(ConversionPatternRewriter &rewriter, Location loc,
 
   return castIntToIndex(rewriter, loc, boundedByDimSize);
 }
+
+// Checks whether the `shapeA` and `shapeB` are broadcast compatible or not. If
+// yes, then computes the final broadcast shape.
+LogicalResult computeBroadcastShape(ConversionPatternRewriter &rewriter,
+                                    Location loc, Value inputA, Value inputB,
+                                    SmallVector<int64_t> &resultShape,
+                                    SmallVector<Value> &resultShapeValue) {
+  SmallVector<int64_t> shapeA{
+      inputA.getType().cast<BaseTensorType>().getSizes()};
+  SmallVector<int64_t> shapeB{
+      inputB.getType().cast<BaseTensorType>().getSizes()};
+  unsigned rankA = shapeA.size();
+  unsigned rankB = shapeB.size();
+  unsigned minRank = rankA > rankB ? rankB : rankA;
+  // Check whether the shapes of the tensors are broadcastable or not.
+  // Two tensors are “broadcastable” if the following rules hold:
+  // 1.) Each tensor has at least one dimension.
+  // 2.) When iterating over the dimension sizes, starting at the trailing
+  // dimension, the dimension sizes must either be equal, one of them is 1, or
+  // one of them does not exist.
+  if (rankA == 0 && rankB == 0) {
+    // Do nothing.
+    return success();
+  } else if (rankA && rankB) {
```

Comment:
I think the code in this section works regardless of whether some of the inputs are zero rank or not. No need for the `if-else` statements